packages/vats/src/core/boot.js

```diff
@@ -107,6 +107,28 @@ const buildRootObject = (vatPowers, vatParameters) => {
     });
 
     await runBehaviors(bootManifest);
+
+    const { coreProposalCode } = vatParameters;
+    if (!coreProposalCode) {
```

Comment:
wish: completely separate boot.js for production and non-production, with no conditionals like this